apps/emqx_modules/src/emqx_delayed.erl

```diff
@@ -378,3 +364,23 @@ do_publish(Key = {Ts, _Id}, Now, Acc) when Ts =< Now ->
 
 -spec delayed_count() -> non_neg_integer().
 delayed_count() -> mnesia:table_info(?TAB, size).
+
+enable(Enable) ->
+    case emqx:get_raw_config([delayed]) of
+        #{<<"enable">> := Enable} ->
+            ok;
+        Cfg ->
+            {ok, _} = update_config(Cfg#{<<"enable">> := Enable}),
+            ok
+    end.
+
+ensure_enable(true, State) ->
+    emqx_hooks:put('message.publish', {?MODULE, on_message_publish, []}),
+    State;
+ensure_enable(false, #{publish_timer := PubTimer} = State) ->
+    emqx_hooks:del('message.publish', {?MODULE, on_message_publish}),
+    emqx_misc:cancel_timer(PubTimer),
+    ets:delete_all_objects(?TAB),
```

Comment:
Test cases are needed for these lines to verify that all messages and hooks are removed?